.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -46,3 +46,21 @@ If this pull request is unrelated to any issues, please remove
 the following line. -->
 
 Fixes #<Issue Number>
+
+### Checklist for package maintainer(s)
+<!-- This section is to be filled by package maintainer(s) who will
+review this pull request. -->
+
+This checklist is meant to remind the package maintainer(s) who will
+review this pull request of some common things to look for. This list
+is not exhaustive.
+
+- [ ] Do the proposed changes actually accomplish desired goals?
+- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
+- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
+- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
+- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
+- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
```

Comment:
How do we know if a failure is related?  Sometimes it's impossible to be sure, as it can be the most bizarre thing...

Maybe suggest as a rule of thumb: Check other recent PR builds on the same CI provider to see if the same failure is occurring.  Not always a 100% guarantee but usually the first thing we check if it's not otherwise clear.

Reply:
Good point. Personally, I check the last `main` run and also see if there are any open issues about it already. Do you want to suggest some wordings to clarify this?

Reply:
I am trying to think of some wording for this, but am struggling to think of anything *succinct* enough that it doesn't just muddle the checklist.

Maybe I can add something about this question to the maintainer guidelines, and the update this checklist later to add a link to it.

Reply:
I created an issue to follow up on this in #11863.